Automattic/wp-calypso
# Improve domain actions permissions with new api properties revised

## Changes proposed in this Pull Request

This PR is an updated version of https://github.com/Automattic/wp-calypso/pull/61963 that takes into account MAPPED domains and solves the issue reported here https://github.com/Automattic/wp-calypso/issues/62109

Depends on D77580-code: apply this diff if it has not been merged yet.

## Testing instructions

Refers to testing instructions described [here](https://github.com/Automattic/wp-calypso/pull/61963), then test it with a mapped domain and verify that DNS card and DNS management page work as they did before.
